Piwik.Tracker/PiwikTracker.cs

```diff
@@ -721,11 +742,11 @@ public TrackingResponse DoTrackEcommerceCartUpdate(double grandTotal)
         /// Response
         /// </returns>
         /// <exception cref="System.InvalidOperationException">Error: you must call the function DoTrackPageView or DoTrackGoal from this class, before calling this method DoBulkTrack()</exception>
-        public TrackingResponse DoBulkTrack()
+        public async Task<TrackingResponse> BulkTrackAsync()

```

Comment:
`async` is required when there is an `await` in the method body.

For the rest of the methods, due to `SendRequestAsync` being the last statement in the method, we could avoid having to await on it and return the `Task` instead (Task is similar to promise in other languages). By doing so, we can also avoid having to decide whether to add `.ConfigureAwait(false)` or not. (`.ConfigureAwait(false)` is always good when no synchronization context is involved, but may cause extra thread switches when synchronization context is needed. See [discussion](https://stackoverflow.com/questions/13489065/best-practice-to-call-configureawait-for-all-server-side-code))